aineko/extras/connectors/secrets.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2023 Aineko Authors
+# SPDX-License-Identifier: Apache-2.0
+"""Secrets utilities."""
+
+import os
+import re
+from typing import Any, Dict, List
+
+
+def _str_inject_secrets(str_: str) -> str:
```

Comment:
It would be helpful to explain what is happening here with the regex and recursion

Reply:
added more documentation